tests/vit_mae/test_modeling_vit_mae.py

```diff
@@ -400,34 +544,31 @@ def default_feature_extractor(self):
 
     @slow
     def test_inference_for_pretraining(self):
-        # make random mask reproducible
-        # note that the same seed on CPU and on GPU doesn’t mean they spew the same random number sequences,
-        # as they both have fairly different PRNGs (for efficiency reasons).
-        # source: https://discuss.pytorch.org/t/random-seed-that-spans-across-devices/19735
-        torch.manual_seed(2)
+        # make random mask reproducible across the PT and TF model
+        np.random.seed(2)
```

Comment:
For this change, it would be great to hear from @NielsRogge too.

Previously we only saw the issue caused by CPU/GPU PRGNs (for the PyTorch ViTMAE).
But now we also see the issue caused by the different frameworks.

I think using `numpy` is a good choice to address this issue. However, let's wait Niels' opinion.

Reply:
It was discussed with @NielsRogge beforehand, though. 

But I don't think this PR is contingent on this change as it was all well discussed before and reviewed as well. I might wrong though.

Reply:
I know we discussed on Slack. As far as I know, Niels knows we are going to use the new `noise` argument and he is happy with this change.

However, regarding using `numpy` instead of `torch` (and updating the expected slice), I don't remember if Niels is aware of this change and expressed his opinion. I might be wrong (?).

Regarding `But now we also see the issue caused by the different frameworks.`, it is just about we decided to use `numpy` because the PRNGs from torch and tf will give different results with the same seed, and we don't want to import torch in tf (non pt-tf cross) test.

Reply:
Thanks for clarifying! 

> because the PRNGs from torch and tf will give different results even with the same seed, 

I think you probably meant TF PT PRNGs won't produce same results even if they were seeded with the same number?

Reply:
yes, that's what I mean.

Reply:
Once Niels is OK with using NumPy, I think it is better to have the comment like

```
# make random mask reproducible across different frameworks (PyTorch, TensorFlow, etc.), as well as across different accelerators (CPU, GPU, etc.).
# note that the same seed on CPU and on GPU doesn’t mean they spew the same random number sequences,
# as they both have fairly different PRNGs (for efficiency reasons).
# source (for PyTorch): https://discuss.pytorch.org/t/random-seed-that-spans-across-devices/19735
```

Reply:
Much more comprehensive. Thanks for suggesting 👌

Reply:
So Niels is OK with this change, I will just let you update the comment, @sayakpaul . Thank you!